src/Time/Units.hs

```diff
@@ -302,10 +301,10 @@ floorUnit = time . floor
 
 -}
 toUnit :: forall (unitTo :: Rat) (unitFrom :: Rat) .
-          (KnownRat unitTo, KnownRat unitFrom, KnownRat (unitFrom / unitTo))
+          (KnownRat unitTo, KnownRat unitFrom)
        => Time unitFrom
        -> Time unitTo
-toUnit Time{..} = Time $ unTime * ratVal (divRat (Proxy @unitFrom) (Proxy @unitTo))
+toUnit Time{..} = Time $ unTime * withDivRat @unitFrom @unitTo (ratVal @(unitFrom / unitTo))
```

Comment:
I did not realize that this was the only use case for `withDivRat`. Why not:

```
toUnit (Time t) = Time (t * ratVal @unitFrom / ratVal @unitTo)
```

and then the `withDivRat` machinery won't be necessary?

Reply:
@int-index The idea I had is that `unitFrom / unitTo` division is performed during compile-time, so when you convert `Week` to `Day` you only multiple `unTime * (5 :%1)` in runtime instead of `unTime * (BIG_NUMBER :% 1) / (ANOTHER_BIG_NUMBER :% 1)`.

Reply:
Then the `reflection`-style solution we have obviously won't work, it performs division at runtime too. We'll have to keep the redundant constraints so that GHC reduces them at compile time.

Reply:
@int-index Ok, I see. So we really need GHC plugin to infer those constraints in order to do this divistion magic at compile-time. 

The current problem we have is that we can't write `toUnit @from @to . toUnit @to @from ≡ id` testing property easily when we have such constraint in type.

Maybe you can suggest a way to implement this property and have `DivRat` constraint in `toUnit` function?

If we implement `toUnit` in simpler way, it means, that we don't need `Gcd`, type level division and all that stuff we did. This makes me feel useless and all our work kinda useless as well :disappointed:

Reply:
>  So we really need GHC plugin to infer those constraints in order to do this divistion magic at compile-time.

No, the plugin would do the same thing as `reflection`, just without an explicit combinator like `withDivRat`.

> Maybe you can suggest a way to implement this property and have DivRat constraint in toUnit function?

Use `withDivRat` in the test, but not in `toUnit`?

Reply:
Oh, and rename `withDivRat` to something like `withRuntimeDivRat`, so it's clear that there's a potential performance penalty.

Reply:
> Use withDivRat in the test, but not in toUnit?

Yes, I realized this just after I wrote this message :slightly_smiling_face: So `withDivRat` combinator is useful. We can use it in tests. But type of `toUnit` and other functions should remain as it was before. Thus, we keep `DivRat` constraints in our library functions.

Reply:
Yeah, this plan sounds good to me.